src/blockdata/script.rs

```diff
@@ -122,30 +122,34 @@ pub enum Error {
     /// Tried to read an array off the stack as a number when it was more than 4 bytes
     NumericOverflow,
     /// Error validating the script with bitcoinconsensus library
-    #[cfg(feature = "bitcoinconsensus")]
-    #[cfg_attr(docsrs, doc(cfg(feature = "bitcoinconsensus")))]
-    BitcoinConsensus(bitcoinconsensus::Error),
+    BitcoinConsensus(BitcoinConsensusError),
     /// Can not find the spent output
-    #[cfg(feature = "bitcoinconsensus")]
-    #[cfg_attr(docsrs, doc(cfg(feature = "bitcoinconsensus")))]
     UnknownSpentOutput(OutPoint),
     /// Can not serialize the spending transaction
-    #[cfg(feature = "bitcoinconsensus")]
-    #[cfg_attr(docsrs, doc(cfg(feature = "bitcoinconsensus")))]
     SerializationError
 }
 
+#[derive(PartialEq, Eq, PartialOrd, Ord, Hash, Debug, Clone, Copy)]
+enum Uninhabited {}
+
+#[cfg(feature = "bitcoinconsensus")]
+type BitcoinConsensusError = bitcoinconsensus::Error;
+
+/// Dummy error type used when `bitcoinconsensus` feature is not enabled.
+#[cfg(not(feature = "bitcoinconsensus"))]
+#[derive(PartialEq, Eq, PartialOrd, Ord, Hash, Debug, Clone, Copy)]
+pub struct BitcoinConsensusError {
+    _uninhabited: Uninhabited,
+}
+
 impl fmt::Display for Error {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match *self {
             Error::NonMinimalPush => write!(f, "non-minimal datapush"),
             Error::EarlyEndOfScript => write!(f, "unexpected end of script"),
             Error::NumericOverflow => write!(f, "numeric overflow (number on stack larger than 4 bytes)"),
-            #[cfg(feature="bitcoinconsensus")]
-            Error::BitcoinConsensus(ref _n) => write!(f, "bitcoinconsensus verification failed"),
-            #[cfg(feature="bitcoinconsensus")]
-            Error::UnknownSpentOutput(ref _point) => write!(f, "unknown spent output Transaction::verify()"),
-            #[cfg(feature="bitcoinconsensus")]
+            Error::BitcoinConsensus(ref err) => write!(f, "bitcoinconsensus verification failed: {:?}", err),
```

Comment:
In d9d27556eacd91923d94571d92c756ee625185ec:

I think we're not supposed to prefix errors, instead using `source`/`cause` for them to let smart error formatters do smart things with them. But I won't hold up this PR -- probably it needs to wait for the MSRV bump so that we can use `source` and do a more general error cleanup.